app/cdap/components/CreateAndEditRecipeForm/index.tsx

```diff
@@ -0,0 +1,202 @@
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React, { FormEvent, useEffect, useState } from 'react';
+import { FormControl } from '@material-ui/core';
+import T from 'i18n-react';
+import {
+  ICreateAndEditRecipeFormProps,
+  IRecipeData,
+} from 'components/CreateAndEditRecipeForm/types';
+import {
+  FormFieldWrapper,
+  NormalLabel,
+  StyledForm,
+  StyledRecipeNameTextField,
+  StyledDescriptionTextArea,
+  StyledCancelButton,
+  StyledSaveButton,
+  StyledFormButtonWrapper,
+  getLabelStyle,
+} from 'components/CreateAndEditRecipeForm/styledComponents';
+import { getCurrentNamespace } from 'services/NamespaceStore';
+import MyDataPrepApi from 'api/dataprep';
+
+export default function({
+  recipeData,
+  setIsCreateAndEditRecipeFormOpen,
+  setSnackbar,
+  recipeFormAction,
+}: ICreateAndEditRecipeFormProps) {
+  const EditRecipeFormAction = 'editRecipe';
+  const [isNameError, setIsNameError] = useState(false);
+  const StyledLabel = getLabelStyle(isNameError);
+  const [recipeFormData, setRecipeFormData] = useState<IRecipeData>({
+    recipeName: '',
+    description: '',
+    directives: [],
+  });
+
+  const [isSaveDisable, setIsSaveDisable] = useState<boolean>(true);
+
+  const recipeSteps = ['uppercase: body1', 'titlecase: body2'];
+
+  useEffect(() => {
+    if (isNameError) {
+      setIsSaveDisable(true);
+    }
+  }, [isNameError]);
+
+  useEffect(() => {
+    setRecipeFormData(recipeData);
+  }, [recipeData]);
+
+  useEffect(() => {
+    if (
+      recipeFormData.recipeName === '' ||
+      recipeFormData.description === '' ||
+      recipeFormData.recipeName?.trim().length === 0 ||
+      recipeFormData.description?.trim().length === 0
+    ) {
+      setIsSaveDisable(true);
+    } else {
+      setIsNameError(false);
+      setIsSaveDisable(false);
+    }
+  }, [recipeFormData]);
+
+  const onCancel = () => {
+    setIsCreateAndEditRecipeFormOpen(false);
+    setIsNameError(false);
+  };
+
+  const onFormHandle = (event: FormEvent) => {
+    event.preventDefault();
+    onRecipeDataSave(recipeFormData);
+  };
+
+  const onRecipeDataSave = (recipeFormData: IRecipeData) => {
+    if (recipeFormAction === EditRecipeFormAction) {
+      const params = {
+        context: getCurrentNamespace(),
+        recipe_id: recipeData.recipeId.recipeId,
+      };
+      const payload = {
+        recipeName: recipeFormData.recipeName,
+        description: recipeFormData.description,
+        directives: recipeFormData.directives,
+      };
+      MyDataPrepApi.updateRecipe(params, payload).subscribe(
```

Comment:
Good to move API calls to separate util or helper file, so that can be used from other Recipe related UI components too.

Reply:
moved API calls to seperate file